src/control/cmd/agent/main.go

```diff
@@ -57,17 +57,69 @@ type cliOptions struct {
 	Insecure   bool       `short:"i" long:"insecure" description:"have agent attempt to connect without certificates"`
 	RuntimeDir string     `short:"s" long:"runtime_dir" description:"Path to agent communications socket"`
 	LogFile    string     `short:"l" long:"logfile" description:"Full path and filename for daos agent log file"`
+	NetworkDbg bool       `short:"n" long:"netdebug" description:"Enable extended network device scan debug output"`
 	Version    versionCmd `command:"version" description:"Print daos_agent version"`
+	NetScan    netScanCmd `command:"scan" description:"Perform local fabric scan"`
 }
 
 type versionCmd struct{}
+type netScanCmd struct {
+	Debug          bool   `short:"d" long:"debug" description:"Enable debug output"`
+	FabricProvider string `short:"p" long:"provider" description:"Filter device list to those that support the given OFI provider (default is all providers)"`
+}
 
 func (cmd *versionCmd) Execute(_ []string) error {
 	fmt.Printf("daos_agent version %s\n", build.DaosVersion)
 	os.Exit(0)
 	return nil
 }
 
+func (cmd *netScanCmd) Execute(args []string) error {
+	var provider string
+	defer os.Exit(0)
+	log := logging.NewCommandLineLogger()
+
+	if cmd.Debug {
+		log.WithLogLevel(logging.LogLevelDebug)
+		netdetect.SetLogger(log)
+	}
+
+	numaAware, err := netdetect.NumaAware()
+	if err != nil {
+		exitWithError(log, err)
+		return nil
+	}
+
+	if !numaAware {
+		fmt.Printf("This system is not NUMA aware")
```

Comment:
These `Printf()`s should be `log.Info()`, no?

Also, while I think this is useful information, I could see an admin responding to that with "Yes, and?"  ... Maybe "No NUMA information available -- All devices will be associated with Socket 0" or something like that.

Reply:
Changed to log.Info() and updated the message.